src/Workspaces/Core/Portable/Remote/RemoteArguments.cs

```diff
@@ -91,6 +91,58 @@ public bool Equals(SerializableSymbolAndProjectId other)
         }
     }
 
+    internal class SerializableSymbolUsageInfo : IEquatable<SerializableSymbolUsageInfo>
+    {
+        public bool UsageInfoIsValue;
+        public string UsageInfoString;
+
+        public static SerializableSymbolUsageInfo Dehydrate(SymbolUsageInfo symbolUsageInfo)
+        {
+            bool usageInfoIsValue;
+            string usageInfoString;
+            if (symbolUsageInfo.ValueUsageInfoOpt.HasValue)
+            {
+                usageInfoIsValue = true;
+                usageInfoString = symbolUsageInfo.ValueUsageInfoOpt.Value.ToString();
+            }
+            else
+            {
+                usageInfoIsValue = false;
+                usageInfoString = symbolUsageInfo.TypeOrNamespaceUsageInfoOpt.Value.ToString();
+            }
+
+            return new SerializableSymbolUsageInfo
+            {
+                UsageInfoIsValue = usageInfoIsValue,
+                UsageInfoString = usageInfoString
+            };
+        }
+
+        public SymbolUsageInfo Rehydrate()
+        {
+            return UsageInfoIsValue
```

Comment:
for enum, cheaper way is just casting enum to int and cast int back to enum rather than Enum.Parsing which is expensive.

Reply:
Good point, will change.